plugin-server/src/worker/vm/manager.ts

```diff
@@ -0,0 +1,34 @@
+import { Hub } from 'types'
+
+import { LazyPluginVM } from './lazy'
+
+export class LazyPluginVmManager {
+    vmsList: LazyPluginVM[]
+    nextVmIndex: number
+    stateless: boolean
+
+    constructor(server: Hub, stateless = false) {
+        this.stateless = stateless
+        this.nextVmIndex = 0
+        this.vmsList = []
+        const maxVms = stateless ? server.TASKS_PER_WORKER * 2 : 1
```

Comment:
Fly-by because I'm not familiar enough with plugin server...

I like this approach because of the direction it sets

* AWS Lambda allow a certain amount of concurrency (1000 concurrent functions per account by default)
* people with certain permissions can choose to specify allowed concurrency (e.g. function A must never fail so reserve X running functions for it) or can leave it for the platform to manage
* if the account hits the limit function invocations will queue and might ultimately time out
* you can contact your account manager to raise the limit (which means they get to ask you why)
* you can pay to provision concurrency

So, I'm generally in favour of introducing the concept of configurable numbers of invocations/workers